plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -202,14 +275,10 @@ public void onResponse(MLTaskResponse mlTaskResponse) {
                         // document field as key, model field as value
                         String newDocumentFieldName = entry.getKey();
                         String modelOutputFieldName = entry.getValue();
-                        if (ingestDocument.hasField(newDocumentFieldName)) {
-                            throw new IllegalArgumentException(
-                                "document already has field name "
-                                    + newDocumentFieldName
-                                    + ". Not allow to overwrite the same field name, please check output_map."
-                            );
+                        if (!newOutputMapping.containsKey(newDocumentFieldName)) {
```

Comment:
please write unit test to test onFailure  and rest the failure

Reply:
working on unit tests now